File: client/csharp/example/client/GrpcClient.cs
using System;
using System.IO;
using Grpc.Core;

namespace example
{
  class GrpcClient
  {
    private Channel channel;

    public void Connect(String caFile, String serverAddr, int serverPort) {
      var channelCredentials = new SslCredentials(File.ReadAllText(caFile));

      channel = new Channel(serverAddr, serverPort, channelCredentials);
    } 

    public Channel GetChannel() {
      return channel;
    }

    public void Close() {
      channel.ShutdownAsync().Wait();
    }
  }
}

File: client/java/src/main/java/com/supremainc/sdk/example/client/GrpcClient.java
package com.supremainc.sdk.example.client;

import io.grpc.ManagedChannel;
import io.grpc.netty.NettyChannelBuilder;
import io.grpc.netty.GrpcSslContexts;

import java.util.concurrent.TimeUnit;

import java.io.File;

public class GrpcClient {
  private ManagedChannel channel;

  public ManagedChannel getChannel() {
    return channel;
  }

  public void connect(String certFile, String serverAddr, int serverPort) throws Exception {
    channel = NettyChannelBuilder.forAddress(serverAddr, serverPort)
    .sslContext(GrpcSslContexts.forClient().trustManager(new File(certFile)).build())
    .build();
  }

  public void close() throws InterruptedException {
    if(channel != null) {
      channel.shutdown().awaitTermination(5, TimeUnit.SECONDS);
      channel = null;
    }
  }
}

